src/ucp/wireup/wireup.c

```diff
@@ -1012,7 +1016,7 @@ ucp_wireup_check_config_intersect(ucp_ep_h ep,
     ucp_ep_config_lanes_intersect(old_key, new_key, reuse_lane_map);
 
     for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
-        ucs_assert(ep->uct_eps[lane] != NULL);
+        ucs_assert((ep->uct_eps[lane] != NULL) || (lane == old_key->cm_lane));
```

Comment:
why needed? seems it's already after restore from tmp_ep

Reply:
no, it goes from change at line 458, since we delay restore EP, we have to initialize lanes on TMP EP

Reply:
and on tmp_ep, the cm_lane is NULL because we don't bother initializing it?
can you pls add (ep->flags&TEMP) to the assert?
e.g
assert(lane!=NULL) || (ep->flags&TEMP && lane==oldkey->cm_lane)
and add comment to the code?